src/pytorch_lightning/tuner/tuning.py

```diff
@@ -62,14 +69,38 @@ def _tune(
         if self.trainer.auto_scale_batch_size:
             if isinstance(self.trainer.auto_scale_batch_size, str):
                 scale_batch_size_kwargs.setdefault("mode", self.trainer.auto_scale_batch_size)
-            result["scale_batch_size"] = scale_batch_size(self.trainer, model, **scale_batch_size_kwargs)
+
+            result["scale_batch_size"] = self.scale_batch_size(
+                model, train_dataloaders, val_dataloaders, dataloaders, datamodule, method, **scale_batch_size_kwargs
+            )
 
         # Run learning rate finder:
         if self.trainer.auto_lr_find:
+            self.trainer.state.fn = TrainerFn.TUNING
+            self.trainer.state.status = TrainerStatus.RUNNING
+            self.tuning = True
+
+            # TODO: Remove this once LRFinder is converted to a Callback
+            # if a datamodule comes in as the second arg, then fix it for the user
+            if isinstance(train_dataloaders, LightningDataModule):
+                datamodule = train_dataloaders
+                train_dataloaders = None
+
+            # If you supply a datamodule you can't supply train_dataloader or val_dataloaders
+            if (train_dataloaders is not None or val_dataloaders is not None) and datamodule is not None:
+                raise MisconfigurationException(
+                    "You cannot pass `train_dataloader` or `val_dataloaders` to `trainer.tune()`"
+                    " if datamodule is already passed."
+                )
```

Comment:
Can you add support for TUNING in attach_data instead? As in https://github.com/Lightning-AI/lightning/pull/14637

Reply:
it's removed in the LRfinder PR. We won't need it.